jwt.go

```diff
@@ -179,18 +182,72 @@ func (ja *JWTAuth) usingJWK() bool {
 }
 
 func (ja *JWTAuth) setupJWKLoader() {
+	// Initialiser le cache pour toutes les URL possibles
+	ja.jwkCaches = make(map[string]*struct {
+		cache     *jwk.Cache
+		cachedSet jwk.Set
+	})
+	ja.logger.Info("JWK cache initialized for placeholder URL", zap.String("placeholder_url", ja.JWKURL))
+}
+
+// getOrCreateJWKCache retrieves or creates a cache for a specific JWK URL
+func (ja *JWTAuth) getOrCreateJWKCache(resolvedURL string) (*struct {
+	cache     *jwk.Cache
+	cachedSet jwk.Set
```

Comment:
Please simplify this by using a new type, which is requested in my comments above.